dvc/command/stage.py

```diff
@@ -194,3 +292,47 @@ def add_parser(subparsers, parent_parser):
     )
     _add_common_args(stage_add_parser)
     stage_add_parser.set_defaults(func=CmdStageAdd)
+
+    STAGE_LIST_HELP = "List stages."
+    stage_list_parser = stage_subparsers.add_parser(
+        "list",
+        parents=[parent_parser],
+        description=append_doc_link(STAGE_LIST_HELP, "stage/list"),
+        help=STAGE_LIST_HELP,
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+    )
+    stage_list_parser.add_argument(
+        "targets",
+        nargs="*",
+        default=["dvc.yaml"],
+        help=(
+            "Show stages from a dvc.yaml/.dvc file or a directory. "
+            "'dvc.yaml' by default"
+        ),
+    )
+    stage_list_parser.add_argument(
+        "--all",
+        action="store_true",
+        default=False,
+        help="List all of the stages in the repo.",
+    )
+    stage_list_parser.add_argument(
+        "--fail",
+        action="store_true",
+        default=False,
+        help="Fail immediately if there's an error.",
```

Comment:
Doesn't error imply failing already?
Not sure how to try this locally, what kind of error? Thanks

Reply:
Syntax Errors in yaml

Reply:
OK. So my point was that "fails if there's an error" doesn't explain much. Maybe we can update these messages in some upcoming PR?

Reply:
I have opened https://github.com/iterative/dvc/issues/5390 regarding discussion of UI messages. Will update the flags as that will most likely affect it.

Reply:
Yeah I found that issue after checking this. Cool, thanks!

Reply:
Tried to address this on #5499, please take a look.

Reply:
#5499 LGTM 👍